### PR TITLE
Compiler benchmark

### PR DESCRIPTION
**Pull request by bmsykes**
- Added a benchmark job
- Compiles with both gnu and intel and runs the discF_00500 test
- Timing information is output through the Actions log

### CODE REVIEW

File: .github/actions/benchmark/action.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: "benchmark"
+description: "run performance benchmark (for macos only)"
+
+# TODO: if any other runs fail then don't save benchmark results (if the benchmark somehow runs e.g. linux-only bug)
```
Comment (conradtchan) [inline .github/actions/benchmark/action.yml:4]: Looks like this has been implemented. Is it okay to remove this TODO comment?

**Review by conradtchan — CHANGES_REQUESTED** [review]
Aside from the comment, this looks good to me 👍

File: .github/actions/benchmark/action.yml (hunk 1 above)
Comment (bmsykes) [reply .github/actions/benchmark/action.yml:4]: Yep!

**Review by conradtchan — APPROVED** [review]

**Review by dliptai — APPROVED** [review]

**Comment by conradtchan** [comment]
@cpinte This PR is ready, please feel free to merge it in if you're happy with the changes 🙂

**Review by cpinte — APPROVED** [review]

**Comment by cpinte** [comment]
Thanks !

**Pull request merged**